benchmarks/python/benchmark_overlap.py

```diff
@@ -0,0 +1,243 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import pytest
+import torch
+import nvfuser
+from nvfuser import FusionDefinition, CommunicatorBackend
+from nvfuser.pytorch_utils import torch_dtype_to_nvfuser_dtype
+from .core import BENCHMARK_CONFIG, clear_l2_cache
+
+
+class CUDAEventTimer:
+    """Custom CUDA event-based timer for accurate GPU timing.
+
+    This timer uses CUDA events to measure elapsed time between operations,
+    providing more accurate GPU timing than CPU-based timing methods.
+    """
+
+    def __init__(self):
+        self.start_event = torch.cuda.Event(enable_timing=True)
+        self.end_event = torch.cuda.Event(enable_timing=True)
+        self.is_running = False
+
+    def __call__(self):
+        """Record timing events and compute elapsed time.
+
+        Returns:
+            float: Elapsed time in seconds
+        """
+        if self.is_running:
+            self.end_event.record()
+            torch.cuda.synchronize()
+            elapsed_ms = self.start_event.elapsed_time(self.end_event)
+            self.is_running = False
+            return elapsed_ms / 1000.0  # Convert ms to seconds
+        else:
+            self.start_event.record()
+            self.is_running = True
+            return 0.0
+
+    def cleanup(self):
+        """Ensure timer is not running and synchronize CUDA."""
+        if self.is_running:
+            self.end_event.record()
+            torch.cuda.synchronize()
+            self.is_running = False
+
+
+def benchmark_cuda_events_pedantic(
+    benchmark, benchmark_fn, inputs, rounds, warmup_rounds
+):
+    """Wrapper for benchmark_cuda_events that uses pytest's pedantic method with CUDA events.
+
+    Args:
+        benchmark: pytest-benchmark fixture
+        benchmark_fn: Function to benchmark
+        inputs: List of inputs to pass to benchmark_fn
+        rounds: Number of rounds to run
+        warmup_rounds: Number of warmup rounds
+    """
+
+    def setup():
+        clear_l2_cache()
+        return inputs, {}
+
+    def wrapped_fn(*args):
+        benchmark_fn(*args[0])
+        return None
+
+    # Set our custom CUDA event timer
+    benchmark._timer = CUDAEventTimer()
```

Comment:
Did you run into issues around the precision of timer?
If no timer precision is set, pytest-benchmark does its own calibration which can sometimes produce invalid results: https://github.com/NVIDIA/Fuser/blob/c229fd5602775529022d4d5b116ee0a3818d061b/benchmarks/python/core.py#L112-L115

Reply:
no I didn't get any issue about the precision of timer. Do you want me to manually set the precision anyway ?